include/mbedtls/cmac.h

```diff
@@ -35,104 +36,118 @@ extern "C" {
+     *  pending processing, or the final block. */
     unsigned char       unprocessed_block[MBEDTLS_CIPHER_BLKSIZE_MAX];
 
-    /** Length of data pending to be processed */
+    /** The length of data pending processing. */
     size_t              unprocessed_len;
 };
 
 /**
- * \brief               Set the CMAC key and prepare to authenticate the input
- *                      data.
- *                      Should be called with an initialized cipher context.
- *
- * \param ctx           Cipher context. This should be a cipher context,
- *                      initialized to be one of the following types:
- *                      MBEDTLS_CIPHER_AES_128_ECB, MBEDTLS_CIPHER_AES_192_ECB,
- *                      MBEDTLS_CIPHER_AES_256_ECB or
- *                      MBEDTLS_CIPHER_DES_EDE3_ECB.
- * \param key           CMAC key
- * \param keybits       length of the CMAC key in bits
- *                      (must be acceptable by the cipher)
- *
- * \return              0 if successful, or a cipher specific error code
+ * \brief               This function sets the CMAC key, and prepares to authenticate 
+ *                      the input data.
+ *                      Must be called with an initialized cipher context.
+ *
+ * \param ctx           The context of the cipher, initialized as one of the 
+ *                      following types:<ul>
+ *                      <li>MBEDTLS_CIPHER_AES_128_ECB</li>
+ *                      <li>MBEDTLS_CIPHER_AES_192_ECB</li>
+ *                      <li>MBEDTLS_CIPHER_AES_256_ECB</li>
+ *                      <li>MBEDTLS_CIPHER_DES_EDE3_ECB</li></ul>
+ * \param key           The CMAC key.
+ * \param keybits       The length of the CMAC key in bits.
+ *                      Must be supported  by the cipher.
+ *
+ * \return              \c 0 on success, or a cipher-specific error code.
  */
 int mbedtls_cipher_cmac_starts( mbedtls_cipher_context_t *ctx,
                                 const unsigned char *key, size_t keybits );
 
 /**
- * \brief               Generic CMAC process buffer.
- *                      Called between mbedtls_cipher_cmac_starts() or
- *                      mbedtls_cipher_cmac_reset() and
- *                      mbedtls_cipher_cmac_finish().
- *                      May be called repeatedly.
- *
- * \param ctx           CMAC context
- * \param input         buffer holding the  data
- * \param ilen          length of the input data
- *
- * \returns             0 on success, MBEDTLS_ERR_MD_BAD_INPUT_DATA if parameter
- *                      verification fails.
+ * \brief               This function performs a CMAC operation on  
+ *                      the input buffer.
+ *
+ *                      It is called between mbedtls_cipher_cmac_starts() or
+ *                      mbedtls_cipher_cmac_reset(), and mbedtls_cipher_cmac_finish().
+ *                      Can be called repeatedly.
+ *
+ * \param ctx           The CMAC context.
```

Comment:
Suggestion: This is a confusing pre-existing defect in the documentation. What is passed it *not* an instance of `mbedtls_cmac_context_t` (the most natural thing to expect when reading "The CMAC context"), but instead of `mbedtls_cipher_context_t` - indeed, as mentioned above, `mbedtls_cmac_context_t` is supposed to be hidden from the user and only visible for technical reasons. I'd prefer to say "The cipher context used for the CMAC operation." everywhere. @gilles-peskine-arm @Patater @RosZad What do you think?

Reply:
Is this true for all other ciphers/algorithms (I assume yes) - that is: The cipher context used for the <CIPHER/ALGORITHM> operation.

Let's wait to see what Gilles and Jaeden think and if so, I will change in all files consistently.

Reply:
@RosZad This is a peculiarity of CMAC, the other ciphers aren't implemented this way.

I concur with Hanno, “CMAC context” is misleading, the documentation should say “the cipher context used for the CMAC operation” instead.

This applies to all the parameters `mbedtls_cipher_context_t *ctx` of functions in this module.

Reply:
fixed